packages/alchemy/src/provider.ts

```diff
@@ -1,33 +1,35 @@
 import {
   BaseSmartContractAccount,
   SmartAccountProvider,
+  createPublicErc4337Client,
   deepHexlify,
   resolveProperties,
   type AccountMiddlewareFn,
   type SmartAccountProviderOpts,
 } from "@alchemy/aa-core";
-import type { Address, Chain, HttpTransport } from "viem";
+import { type Address, type Chain, type HttpTransport } from "viem";
 import {
   arbitrum,
   arbitrumGoerli,
   optimism,
   optimismGoerli,
 } from "viem/chains";
 import { SupportedChains } from "./chains.js";
+import type { ClientWithAlchemyMethods } from "./middleware/client.js";
+import { withAlchemyGasFeeEstimator } from "./middleware/gas-fees.js";
 import {
+  alchemyPaymasterAndDataMiddleware,
   withAlchemyGasManager,
   type AlchemyGasManagerConfig,
-  alchemyPaymasterAndDataMiddleware,
 } from "./middleware/gas-manager.js";
-import { withAlchemyGasFeeEstimator } from "./middleware/gas-fees.js";
-import type { ClientWithAlchemyMethods } from "./middleware/client.js";
 
 type ConnectionConfig =
   | {
       apiKey: string;
-      rpcUrl?: undefined;
+      rpcUrl?: never;
     }
-  | { rpcUrl: string; apiKey?: undefined };
+  | { rpcUrl: string; apiKey?: never }
```

Comment:
oh right, I ran into a typing thing if I didn't also have `jwt?: never` in all the other options. Then you can do `connectionConfig.jwt != null ? {} : {headers: ...}` down below

I wonder if this causes any problems elsewhere re typing. probs not